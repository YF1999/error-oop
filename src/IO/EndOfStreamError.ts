import { ErrorTool, IErrorArguments } from '../Native';

import { IOError, IIOErrorOptions } from './IOError';

export interface IEndOfStreamErrorOptions extends IIOErrorOptions {}

/**
 * Applicable when reading is attempted past the end of a stream.
 */
export class EndOfStreamError extends IOError {
    public constructor();
    /**
     * @param message The error message that explains the reason for this error.
     */
    public constructor(message: string);
    /**
     * @param message The error message that explains the reason for this error.
     * @param innerError The error that is the cause of the current error. Stack trace will be appended.
     */
    public constructor(message: string, innerError: Error);
    /**
     * @param options The constructor options.
     */
    public constructor(options: IEndOfStreamErrorOptions);

    public constructor(...args: IErrorArguments<IEndOfStreamErrorOptions>) {
        super(ErrorTool.parseErrorArguments([], ...args));
    }
}
