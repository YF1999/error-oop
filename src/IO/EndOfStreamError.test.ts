import { EndOfStreamError } from './EndOfStreamError';

standardTestError(EndOfStreamError, 'EndOfStreamError', 'error');
