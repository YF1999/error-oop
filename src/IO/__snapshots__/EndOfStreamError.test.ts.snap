// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test \`EndOfStreamError\` own keys 1`] = `
Array [
  "stack",
]
`;

exports[`Test \`EndOfStreamError\` own keys 2`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`EndOfStreamError\` own keys 3`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`EndOfStreamError\` own keys 4`] = `
Array [
  "stack",
  "message",
]
`;
